org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/correction/GetterSetterQuickFixTest.java

```diff
@@ -0,0 +1,423 @@
+		IPackageFragment pack = fSourceFolder.createPackageFragment("test", false, null);
+		StringBuilder buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    private byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		ICompilationUnit cu = pack.createCompilationUnit("C.java", buf.toString(), false, null);
+
+		buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		Expected e1 = new Expected("Change visibility of 'test' to 'package'", buf.toString());
+
+		buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    private byte test;\n");
+		buf.append("\n");
+		buf.append("    /**\n");
+		buf.append("     * @return the test\n");
+		buf.append("     */\n");
+		buf.append("    public byte getTest() {\n");
+		buf.append("        return test;\n");
+		buf.append("    }\n");
+		buf.append("\n");
+		buf.append("    /**\n");
+		buf.append("     * @param test the test to set\n");
+		buf.append("     */\n");
+		buf.append("    public void setTest(byte test) {\n");
+		buf.append("        this.test = test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        ++c.test;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		Expected e2 = new Expected("Create getter and setter for 'test'...", buf.toString());
+
+		assertCodeActions(cu, e1, e2);
+	}
+
+	@Test
+	public void testInvisibleFieldToGetterSetter_2() throws Exception {
+		IPackageFragment pack = fSourceFolder.createPackageFragment("test", false, null);
+		StringBuilder buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    private byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        c.test += 1 + 2;\n");
+		buf.append("    }\n");
+		buf.append("}\n");
+		ICompilationUnit cu = pack.createCompilationUnit("C.java", buf.toString(), false, null);
+
+		buf = new StringBuilder();
+		buf.append("package test;\n");
+		buf.append("public class C {\n");
+		buf.append("    byte test;\n");
+		buf.append("}\n");
+		buf.append("class D {\n");
+		buf.append("    public void foo(){\n");
+		buf.append("        C c = new C();\n");
+		buf.append("        c.test += 1 + 2;\n");
```

Comment:
No, that should be `c.setTest(c.getTest() + (1 + 2));`. And you should make `test` an `int` to avoid a compilation error.